src/server/server.ts

```diff
@@ -77,17 +79,53 @@ app.all("/api/*", function (req, res) {
 });
 
 
+
+app.post("/create-chat", function(req, res) {
+  const {sender, to, text} = req.body
+  const chat = new ChatModel({
+    sender,
+    to,
+    text
+  });
+  chat
+  .save()
+  .then((data) => {
+    res.json((data))
+  })
+  .catch((err) => {
+    console.log(err);
+    res.status(501);
+    res.json({errors: err})
+  })
+})
+app.get("/chats", function(req, res) {
+  ChatModel.find()
+  .then((data) => res.json({data}))
+  .catch((err) => {
+    res.status(501);
+    res.json({ errors: err });
+  });
+})
+
 server.listen(PORT, function () {
   console.log(`starting at localhost http://localhost:${PORT}`);
 });
 
 
 io.on('connection', function(socket){
   console.log('a user connected');
-  socket.emit('message', 'work')
-  socket.on('disconnect', function(){
-    console.log('user disconnected');
+ socket.on('join', function(data) {
+   socket.join(data.room)
+   io.emit('new user joined', {user:data.user, message:'joined.'})
+ })
+  socket.on('leave', function(data){
+io.emit('left room', {user:data.user, message:'left room.'});
+socket.leave(data)
   });
+  socket.on('message', function(data) {
+    io.in(data.room).emit('new message', {user:data.user, message:data.essage})
```

Comment:
this is the line that aman requested changes for ... should say data.messages?

Reply:
Good catch. I fixed it.